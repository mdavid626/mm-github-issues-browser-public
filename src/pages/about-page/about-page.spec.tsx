import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Footer from '../../components/footer/footer';
import Header from '../../components/header/header';
import AboutPage from './about-page';

jest.mock('../../components/header/header');
jest.mock('../../components/footer/footer');

describe('about-page', () => {
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    (Header as jest.Mock).mockReturnValue(<div>header</div>);
    (Footer as jest.Mock).mockReturnValue(<div>footer</div>);
    const { asFragment } = render(<AboutPage />);
    expect(asFragment()).toMatchSnapshot();
    expect(Header).toHaveBeenCalledWith({}, {});
    expect(Footer).toHaveBeenCalledWith({}, {});
  });
});
