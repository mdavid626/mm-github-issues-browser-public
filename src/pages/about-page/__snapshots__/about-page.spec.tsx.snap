// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about-page should render 1`] = `
<DocumentFragment>
  <div
    class="AboutPage"
  >
    <div>
      header
    </div>
    <div
      class="AboutPage-content"
    >
      <div
        class="AboutPage-description"
      >
        Simple React app to browse GitHub Issues
      </div>
      <div>
        Created by: 
        <a
          href="https://mdavid626.com"
        >
          Dávid Molnár
        </a>
      </div>
      <a
        href="https://github.com/mdavid626/mm-github-issues-browser"
      >
        Source Code
      </a>
    </div>
    <div>
      footer
    </div>
  </div>
</DocumentFragment>
`;
